UnitTestFrameworkPkg/Test/GoogleTest/Sample/SampleGoogleTest/SampleGoogleTest.cpp

```diff
@@ -294,12 +295,151 @@ TEST (MacroTestsMessages, MacroTraceMessage) {
   ASSERT_TRUE (TRUE);
 }
 
+/**
+  Sample unit test that performs double free
+**/
+TEST (SanitizerTests, DoubleFreeDeathTest) {
+  UINT8  *Pointer;
+
+  Pointer = (UINT8 *)AllocatePool (100);
+  ASSERT_NE (Pointer, (UINT8 *)NULL);
+  FreePool (Pointer);
+  //
+  // Second free that should be caught by address sanitizer, log details, and exit

```

Comment:
It exits the entire GoogleTest or is it able to continue execution outside of this TEST() function?

Reply:
These test cases verify that the address sanitizer does catch the issue and is verified with the use of EXPECT_DEATH() that allows the rest of the tests to be run.

If a developer actually has a double free in their unit test code or code under test, then the address sanitizer would catch the condition, log a stack back trace and other details, and terminate/exit the unit test application.  This is the expected developer experience when a condition is caught by the address sanitizer.  It would also generate an error for CI.